Automattic/pocket-casts-android
# Referrals: gift icon count badge

## Description
This adds referral count badge on the new gift icon. 

Figma: FVJPaWlRMlosEpsRhqxnSD-fi-2189_9583 (there's an open question about screen title)

Part of https://github.com/Automattic/pocket-casts-android/issues/2744

## Testing Instructions
1. Launch the app and login with a paid account
2. Go to Profile screen
3. ✅ Notice that count badge is displayed over the gift icon with max count 3
4. Tap on the icon
5. ✅ Notice that the value decreases and then resets to max count 3 (this is just for demo purpose, actual numbers will be displayed using an API)

## Screenshots or Screencast 

![Untitled](https://github.com/user-attachments/assets/5adde99f-434e-45db-b16b-12cfb8a1c3e2)

![Untitled](https://github.com/user-attachments/assets/d76d1bb4-9f77-456b-9422-0fd0391f42aa)

## Checklist
 
#### I have tested any UI changes...

- [ ] for accessibility with TalkBack - not yet
